# Test generated by RoostGPT for test credential-wallet-rest-assured using AI Type Azure Open AI and AI Model roost-gpt4-32k


# RoostTestHash=b4614c0581

config:
  target: "{{ $processEnvironment.API_HOST }}"
  plugins:
    expect: {}
  phases:
    - duration: 60
      arrivalRate: 2
  payload:
    path: "./persons_post_others.csv"
    fields:
      - "birthDate"
      - "initials"
      - "lastName"
    skipHeader: true

scenarios:
  - name: "Testing non-successful API responses"
    weight: 1
    flow:
      - post:
          url: "/recipients/persons"
          beforeSend: "setHeaders" 
          headers:
            Authorization: "Bearer {{ $processEnvironment.AUTH_TOKEN }}"
            Token: "{{$processEnvironment.AUTH_TOKEN}}"
          json:
            birthDate: "{{ birthDate }}"
            initials: "INCORRECT_VAL"
            lastName: "{{ lastName }}"
          expect:
            - statusCode: 400
            - hasHeader: "api-version"
            - equals:
                error: "validation/initials-pattern"
                description: "Descriptive Error Message"
          think: 1

      - post:
          url: "/UNKNOWN/PATH"
          beforeSend: "setHeaders" 
          headers:
            Token: "{{$processEnvironment.AUTH_TOKEN}}"
          json:
            birthDate: "{{ birthDate }}"
            initials: "INVALID"
            lastName: "{{ lastName }}"
          expect:
            - statusCode: 404
          think: 1

      - post:
          url: "/recipients/persons"
          beforeSend: "setHeaders" 
          headers:
            Token: "INCORRECT_AUTH_TOKEN"
          json:
            birthDate: "{{ birthDate }}"
            initials: "{{ initials }}"
            lastName: "{{ lastName }}"
          expect:
            - statusCode: 401
            - hasHeader: "api-version"
            - equals:
                error: "error/code"
                description: "Descriptive Error Message"
          think: 1
